geode-lucene/src/test/java/org/apache/geode/cache/lucene/test/LuceneFunctionSecurityTest.java

```diff
@@ -0,0 +1,494 @@
+        .statusIsSuccess();
+  }
+
+  @Test
+  @ConnectionConfiguration(user = "noPermissions", password = "noPermissions")
+  public void testInvalidPermissionsForLuceneQueryFunctionWithRegionParameter() throws Exception {
+    Function thisFunction = luceneQueryFunction;
+    ResourcePermission thisRequiredPermission = dataReadRegion;
+
+    gfsh.executeAndAssertThat(
+        "execute function --region=" + regionName + " --id=" + thisFunction.getId())
+        .containsOutput("not authorized for " + thisRequiredPermission.toString())
+        .statusIsSuccess();
+  }
+
+  @Test
+  @ConnectionConfiguration(user = "noPermissions", password = "noPermissions")
+  public void testInvalidPermissionsForWaitUntilFlushedFunctionWithoutRegionParameter()
+      throws Exception {
+    Function thisFunction = waitUntilFlushedFunction;
+    ResourcePermission thisRequiredPermission = dataRead;
+
+    gfsh.executeAndAssertThat("execute function --id=" + thisFunction.getId())
+        .containsOutput("not authorized for " + thisRequiredPermission.toString())
+        .statusIsSuccess();
+  }
+
+  @Test
+  @ConnectionConfiguration(user = "noPermissions", password = "noPermissions")
+  public void testInvalidPermissionsForWaitUntilFlushedFunctionWithRegionParameter()
+      throws Exception {
+    Function thisFunction = waitUntilFlushedFunction;
+    ResourcePermission thisRequiredPermission = dataReadRegion;
+
+    gfsh.executeAndAssertThat(
+        "execute function --region=" + regionName + " --id=" + thisFunction.getId())
+        .containsOutput("not authorized for " + thisRequiredPermission.toString())
+        .statusIsSuccess();
+  }
+
+  @Test
+  @ConnectionConfiguration(user = "noPermissions", password = "noPermissions")
+  public void testInvalidPermissionsForLuceneGetPageFunctionWithoutRegionParameter()
+      throws Exception {
+    Function thisFunction = luceneGetPageFunction;
+    ResourcePermission thisRequiredPermission = dataRead;
+
+    gfsh.executeAndAssertThat("execute function --id=" + thisFunction.getId())
+        .containsOutput("not authorized for " + thisRequiredPermission.toString())
+        .statusIsSuccess();
+  }
+
+  @Test
+  @ConnectionConfiguration(user = "noPermissions", password = "noPermissions")
+  public void testInvalidPermissionsForLuceneGetPageFunctionWithRegionParameter() throws Exception {
+    Function thisFunction = luceneGetPageFunction;
+    ResourcePermission thisRequiredPermission = dataReadRegion;
+
+    gfsh.executeAndAssertThat(
+        "execute function --region=" + regionName + " --id=" + thisFunction.getId())
+        .containsOutput("not authorized for " + thisRequiredPermission.toString())
+        .statusIsSuccess();
+  }
+
+  private static Answer<Void> announceNoOp() {
+    return invocation -> {
+      String mockedName = invocation.getMethod().getName();
+      logger.info("Performing no-op mock of " + mockedName);
```

Comment:
is the logging necessary for testing?

Reply:
It's not necessary, but was useful during iteration of the tests.  It could be removed and the call replaced with a `doNothing()` if you think that would be cleaner and just as intuitive.